### PR TITLE
[DKG integration] do not crash for DKG failures

### PR DESCRIPTION
**Pull request by tarakby**
Addresses [#5931 ](https://github.com/dapperlabs/flow-go/issues/5931)

- add a new sentinel error in the crypto package
- use the new sentinel error in the reactor engine (DKG integration)

### CODE REVIEW

File: crypto/dkg.go [hunk 1]
```diff
@@ -53,6 +58,33 @@ type DKGState interface {
...
+
+// dkgFailureErrorf constructs a new dkgFailureError
+func dkgFailureErrorf(msg string, args ...interface{}) error {
+	return &dkgFailureError{
+		err: fmt.Errorf(msg, args...),
+	}
+}
+
+func (e dkgFailureError) Error() string {
+	return e.err.Error()
```
Comment (jordanschalm) [inline crypto/dkg.go:77]:
```suggestion
	return fmt.Sprintf("dkg failure: %s", e.err.Error())
```
include the context of this error level

**Review by jordanschalm — APPROVED** [review]
Thank you, looks good - one small suggestion.

**Review by kc1116 — APPROVED** [review]

File: crypto/dkg.go (hunk 1 above)
Comment (tarakby) [reply crypto/dkg.go:77]: I think the errors built using `dkgFailureErrorf` are self explanatory. I think wrapping the error message in a context is useful when the error itself is returned by another function, for instance [here](https://github.com/onflow/flow-go/blob/tarak/5931-dkg-failure-bug/engine/consensus/dkg/reactor_engine.go#L278).

On a different note, I think I could embedded the error in the structure `dkgFailureError` and avoid defining `Error()`.

Comment (jordanschalm) [reply crypto/dkg.go:77]: I think it's useful to have an error type include some context for example for searching logs. 

If I want to search for `dkgFailureError` errors, I can search for the "dkg failure: "` string

Comment (tarakby) [reply crypto/dkg.go]: added in https://github.com/onflow/flow-go/pull/1693/commits/52236feb1ee7f0e17337c5129eb951b6b56e55d8 with missing tests

**Comment by tarakby** [comment]
bors merge

**Pull request merged**